lib/src/http/common/util.dart

```diff
@@ -17,7 +17,14 @@ library w_transport.src.http.common.util;
 import 'dart:async';
 import 'dart:convert';
 
-StreamTransformer decodeAttempt(Encoding encoding) {
+StreamTransformer decodeAttempt(Encoding encoding) =>
+    _attemptToApplyEncoding(encoding, decode: true);
+
+StreamTransformer encodeAttempt(Encoding encoding) =>
+    _attemptToApplyEncoding(encoding, encode: true);
+
+StreamTransformer _attemptToApplyEncoding(Encoding encoding,
+    {bool encode: false, bool decode: false}) {
```

Comment:
the use of 2 mutually exclusive bools as params would likely be confusing.  Can you come up with 1 param that still adequately describes the use case? (could be a setting with 2 possible enum values or just a single more descriptive bool)